runtime-sdk/src/callformat.rs

```diff
@@ -196,42 +196,67 @@ pub fn encode_result<C: Context>(
     ctx: &C,
     result: module::CallResult,
     metadata: Metadata,
+) -> CallResult {
+    encode_result_ex(ctx, result, metadata, false /* show_raw_errors */)
+}
+
+pub fn encode_result_ex<C: Context>(
+    ctx: &C,
+    result: module::CallResult,
+    metadata: Metadata,
+    show_raw_errors: bool,
```

Comment:
```suggestion
    expose_failure: bool,
```
I recommend this because the encrypted errors aren't "cooked" but rather repacked/hidden.